### PR TITLE
Initial windows CI that should build release from sources

### PR DESCRIPTION
**Pull request by leshikus**
Subj

### CODE REVIEW

**Comment by gshimansky** [comment]
This PR depends on #431 for successful build.

**Comment by alexbaden** [comment]
@gshimansky can you rebase this on main since #431 was merged?

**Comment by gshimansky** [comment]
> @gshimansky can you rebase this on main since #431 was merged?

Ok, rebased.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -17,6 +17,13 @@ jobs:
     name: Build
     uses: ./.github/workflows/build.yml
 
+  windows-build:
+    name: Build on Windows
+    uses: ./.github/workflows/windows-build.yml
+    with:
+      name: release
+      options: -DCMAKE_BUILD_TYPE=Release -DENABLE_FOLLY=off -G "Visual Studio 17 2022"
```
Comment (kurapov-peter) [inline .github/workflows/main.yml:25]: Does folly not work on windows?

Comment (gshimansky) [reply .github/workflows/main.yml:25]: I suppose it works but there is no conda-forge build on windows
https://github.com/conda-forge/folly-feedstock/blob/main/recipe/meta.yaml#L19

Comment (alexbaden) [reply .github/workflows/main.yml:25]: We decided to skip it for now, which would use std lib atomics on Windows. The Arena allocator is a bigger issue we need to solve.

Comment (gshimansky) [reply .github/workflows/main.yml:25]: Someone is trying to write windows build so maybe eventually they will add one
https://github.com/conda-forge/folly-feedstock/pull/98

**Review by alexbaden — APPROVED** [review]

**Pull request merged**